benchmark/Main.hs

```diff
@@ -1,25 +1,32 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+{-# LANGUAGE DataKinds        #-}
+{-# LANGUAGE KindSignatures   #-}
 {-# LANGUAGE TypeApplications #-}
 
 module Main where
 
-import Gauge (bench, bgroup, defaultMain, whnf)
+import Control.DeepSeq (NFData)
+import Gauge (bench, bgroup, defaultMain, nf, whnf)
 
-import Time (Day, Nanosecond, Second, ns, sec, toUnit, week)
+import Time (Day, Nanosecond, Rat, Second, Time, ns, sec, toUnit, week)
 
 import qualified Data.Time.Units as TU (Day, Nanosecond, Second, Week, convertUnit)
 
+instance NFData (Time (unit :: Rat))
```

Comment:
`NFData` type class has default definition of `rnf` function which gives us behavior similar to `whnf` function. Could you try to use `StandaloneDeriving` language extension and write this instance like this:

```haskell
deriving NFData (Time (unit :: Rat))
```